ARCHITECTURE.md

```diff
@@ -0,0 +1,54 @@
+
+## Back-end
+
+The SkyPortal back-end is built using [Tornado](https://www.tornadoweb.org/en/stable/), a Python web application framework that provides its own I/O event loop for non-blocking sockets, making it ideal for use with websockets.
+
+To handle HTTP requests, we define _request handlers_ that are mapped to API endpoints in the application's configuration (in `skyportal/app_server.py` -- see below). Each SkyPortal request handler is a subclass of
+Handler`(defined in`skyportal/handlers/base.py`), a handler that extends Tornado's base [RequestHandler](https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler), handling authentication and providing utility methods for pushing websocket messages to the front-end and returning HTTP responses.
+
+Handlers appear in the directory `skyportal/handlers/api`.
+
+## Database
+
+SkyPortal uses a PostgreSQL database to manage persistent state. The SkyPortal Python backend interacts with the PostgreSQL backend using the
+`SQLAlchemy <http://sqlalchemy.org>`_
+`object relational mapper <https://docs.sqlalchemy.org/en/20/orm/tutorial.html>`_.
+Each database table is represented by a Python class, and each table Column
+is represented by a class attribute.
+
+The models defined in SkyPortal can be found in the directory `skyportal/models`, while those found in baselayer can be found in `baselayer/app/models.py`. When adding a new model, one of the GitHub actions automatically will create the alembic migration file, which should then be committed in the directory `alembic/versions`.
+
+## Permissions
+
+SkyPortal has permissions on handler interactions. For example, the decorator `@auth_or_token` tells the application that the request must either come from a logged in user (via the browser), or must include a valid token in the request header. If neither of these are true, the request returns with an error. There are also specific permissions for interactions within the app (in `skyportal/model_util.py`), that can be assigned to users and then specified in a handler as follows:
+
+```
+from baselayer.app.access import permissions
+@permissions(["System admin"])
+```
+
+## Testing
+
+SkyPortal has a test suite, which appear in the directory `skyportal/tests`. They are generally broken down into front-end and back-end, however, there is also a `flaky` directory for tests that do not always pass, such as those with external interactions. The [developer guide](./doc/dev.md) contains more information on test infrastructure.
+
+## Plotting
+
+SkyPortal uses [plotly](https://github.com/plotly/plotly.py) for interactive plotting. Examples include the photometry plot (in `/components/PhotometryPlot.jsx`) and spectroscopy plot (in `/components/SpectraPlot.jsx`).
```

Comment:
We also use vega for lightweight simple plots that don't require too much customization

Reply:
Added this too.